### PR TITLE
Sync PaymentProcessor on startup

### LINKED ISSUE #2398: Re-enable local geth sync before starting Golem
This feature was removed in: https://github.com/golemfactory/golem/commit/8502d35a4202cbbe94914eda978268e14ad808ed#diff-9f351727974c5655127c6a4dc0671995R177 
because it was causing too much problems and most users use extrenal geths.

### LINKED ISSUE #2585: Creating new task fails if transaction system doesn't know about funds yet
```
ERROR    [golem.client                       ] Cannot create task {... cut ...}                                                                                                   
Traceback (most recent call last):
  File "/home/etam/code/github/golemfactory/golem/golem/client.py", line 658, in create_task
    task = yield self.enqueue_new_task(t_dict)
  File "/home/etam/code/github/golemfactory/golem/golem/client.py", line 533, in enqueue_new_task
    self.funds_locker.lock_funds(task)
  File "/home/etam/code/github/golemfactory/golem/golem/transactions/ethereum/fundslocker.py", line 62, in lock_funds                                                                                                                             
    if tfl.gnt_lock() > gnt - lock_gnt:
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
```

### PR DESCRIPTION
**Pull request by Krigpl**
Resolves #2398
Fixes #2585

### CODE REVIEW

File: golem/client.py [hunk 1]
```diff
@@ -245,9 +245,8 @@ def taskmanager_listener(self, sender, signal, event='default', **kwargs):
         )
         self._publish(Task.evt_task_status, kwargs['task_id'])
 
-    # TODO: re-enable. issue #2398
     def sync(self):
-        pass
+        self.transaction_system.sync()
```
Comment (mfranciszkiewicz) [inline golem/client.py]: Could you decorate `sync` with `@report_calls(Component.client, 'sync')`? This way we can inform the UI on what's happening and show a message to the user.

**Review by mfranciszkiewicz — APPROVED** [review]

**Comment by etam** [comment]
Why are changes in `golem/node.py` required?

**Comment by Krigpl** [comment]
@etam They aren't really. This is a leftover after I was considering a different approach where RPC calls to client would be enabled only after the client has started (which in general makes sense).

**Comment by etam** [comment]
@Krigpl If so, I think it should be in a separate pull request.

**Comment by Krigpl** [comment]
> If so, I think it should be in a separate pull request.

Very well.

**Review by etam — APPROVED** [review]
approved

**Pull request merged**